harness/determined/estimator/_estimator_context.py

```diff
@@ -34,8 +34,13 @@ class EstimatorContext(estimator._EstimatorReducerContext):
     workflow that uses the ``tf.estimator`` API.
     """
 
-    def __init__(self, env: det.EnvContext, hvd_config: horovod.HorovodContext) -> None:
-        super().__init__()
+    def __init__(
+        self,
+        env: det.EnvContext,
+        hvd_config: horovod.HorovodContext,
+        allgather_fn: Callable[[Any], List[Any]],
```

Comment:
:D This makes the inheritance relationship between `EstimatorContext` and `EstimatorTrialContext` looks dumb since `TrialContext` inherits `TrainContext`, which contains the allgather function. I understand this is just workaround to make the od code path work. So I won't let you refactor the whole class inheritance relationships.

Reply:
Yeah.  I didn't spend too much time refactoring the inheritance pattern here, largely because of the Native Context laying around.  It's only the Native context that even distinguishes between the `EstimatorContext` and the `EstimatorTrialContext`, so after Native is gone there will only be `EstimatorContext`.